packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -5341,6 +5341,27 @@ export class Checker extends ParseTreeWalker {
                         return true;
                     }
 
+                    // if containingClass (that initializes this symbol) is a method that is called in __init__
+                    // then we consider this symbol initialized
+                    if (containingClass.nodeType === ParseNodeType.Function) {
+                        const initNode = ClassType.getSymbolTable(classType).get('__init__')?.getDeclarations().at(0)
+                            ?.node;
+                        if (initNode && initNode.nodeType === ParseNodeType.Function) {
+                            return initNode.d.suite.d.statements.some((maybeStatementList) => {
```

Comment:
Some reviewers might be concerned about the performance impact.

Note that the common normal case will usually short-circuit in the check right above this new code.
`if (containingClass.d.name.d.value === '__init__') {`

So this new code usually won't run unless it's actually valuable.

Reply:
seems fine, looking at the time it took to run the primer the performance impact appears to be minimal